CMakeLists.txt

```diff
@@ -0,0 +1,245 @@
+endif()
+
+sdl_find_sdl2(${sdl2_target_name} ${SDL_REQUIRED_VERSION})
+
+add_library(SDL2_net
+    SDLnet.c
+    SDLnetselect.c
+    SDLnetTCP.c
+    SDLnetUDP.c
+)
+add_library(SDL2_net::${sdl2_net_export_name} ALIAS SDL2_net)
+target_include_directories(SDL2_net PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>"
+)
+target_compile_definitions(SDL2_net PRIVATE
+    BUILD_SDL
+    SDL_BUILD_MAJOR_VERSION=${MAJOR_VERSION}
+    SDL_BUILD_MINOR_VERSION=${MINOR_VERSION}
+    SDL_BUILD_MICRO_VERSION=${MICRO_VERSION}
+)
+target_link_libraries(SDL2_net PRIVATE ${sdl2_target_name})
+if(WIN32) 
+    set_target_properties(SDL2_net PROPERTIES COMPILE_FLAGS "/wd4244 /wd4996")
+    target_compile_definitions(SDL2_net PRIVATE _WINSOCK_DEPRECATED_NO_WARNINGS)
+    target_link_libraries(SDL2_net PRIVATE ws2_32 iphlpapi)
+    if(SDL2NET_BUILD_SHARED_LIBS)
+        target_sources(SDL2_net PRIVATE
+            version.rc
+        )
+    endif()
+endif()
+set_target_properties(SDL2_net PROPERTIES
+    DEFINE_SYMBOL DLL_EXPORT
+    PUBLIC_HEADER SDL_net.h
+    EXPORT_NAME ${sdl2_net_export_name}
+    C_VISIBILITY_PRESET "hidden"
+)
+if(NOT ANDROID)
+    set_target_properties(SDL2_net PROPERTIES
+        DEBUG_POSTFIX "${SDL2NET_DEBUG_POSTFIX}"
+    )
+    if(APPLE)
+        # the SOVERSION property corresponds to the compatibility version and VERSION corresponds to the current version
+        # https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html#mach-o-versions
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${DYLIB_COMPATIBILITY_VERSION}"
+            VERSION "${DYLIB_CURRENT_VERSION}"
+        )
+    else()
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${LT_MAJOR}"
+            VERSION "${LT_VERSION}"
+        )
+    endif()
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS AND (APPLE OR (UNIX AND NOT ANDROID)))
+    add_custom_command(TARGET SDL2_net POST_BUILD
+        COMMAND "${CMAKE_COMMAND}" -E create_symlink "$<TARGET_SONAME_FILE_NAME:SDL2_net>" "libSDL2_net$<$<CONFIG:Debug>:${SDL2NET_DEBUG_POSTFIX}>$<TARGET_FILE_SUFFIX:SDL2_net>"
+        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}"
+    )
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS)
+    if(WIN32 OR OS2)
+        set_target_properties(SDL2_net PROPERTIES
+            PREFIX ""
+        )
+    endif()
+    if(OS2)
+        # FIXME: OS/2 Makefile has a different LIBNAME
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2net"
+        )
+    elseif(UNIX AND NOT ANDROID)
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2_net-${LT_RELEASE}"
+        )
+    endif()
+endif()
+
+# Use `Compatible Interface Properties` to ensure all linked SDL2 libraries are built with the same "sharedness".
+set_target_properties(SDL2_net PROPERTIES 
+    INTERFACE_SDL2_SHARED ${SDL2NET_BUILD_SHARED_LIBS}
+    COMPATIBLE_INTERFACE_BOOL SDL2_SHARED
+)
```

Comment:
I added this to `SDL_image` to avoid linking a static SDL2 and a shared SDL2_image together.
This can happen when vendoring both libraries.
But perhaps this implementation is too limiting: it also forbids a shared SDL2 and a static SDL2_image.

Perhaps the following should be done instead:
in the main sdl project:
```cmake
set_target_properties(SDL2 PROPERTIES 
    INTERFACE_SDL2_SHARED ${SDL_SHARED}
    COMPATIBLE_INTERFACE_BOOL SDL2_SHARED
)
```

and in all satellite libraries:
```cmake
if(SDL2_XXX_BUILD_SHARED_LIBS)
    set_target_properties(SDL2_xxx PROPERTIES
        INTERFACE_SDL2_SHARED ${SDL2NET_BUILD_SHARED_LIBS}
        COMPATIBLE_INTERFACE_BOOL SDL2_SHARED
    )
endif()
```
Then it will only fail in the SDL2=static + SDL2_xxx=shared state.

Reply:
Can you add this?
The same change should happen to `SDL_image`, for the same reason I explained above.
```suggestion
# Use `Compatible Interface Properties` to ensure a shared SDL2_net is built with a shared SDL2
if(SDL2NET_BUILD_SHARED_LIBS)
    set_target_properties(SDL2_net PROPERTIES 
        INTERFACE_SDL2_SHARED ${SDL2NET_BUILD_SHARED_LIBS}
        COMPATIBLE_INTERFACE_BOOL SDL2_SHARED
    )
endif()
```